packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl

```diff
@@ -1,23 +1,27 @@
-import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:{{projectName}}/{{projectName}}.dart';
+import 'package:{{projectName}}/{{projectName}}_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
-void main() {
-  const MethodChannel channel = MethodChannel('{{projectName}}');
+class Mock{{pluginDartClass}}Platform 
+    with MockPlatformInterfaceMixin
+    implements {{pluginDartClass}}Platform {
 
-  TestWidgetsFlutterBinding.ensureInitialized();
+  @override
+  Future<String?> getPlatformVersion() => Future.value('42');
+}
 
+void main() {
+  late {{pluginDartClass}} {{pluginClassLowerCamelCase}};
```

Comment:
There's no reason for the object under test to be part of the fixture, and it's a practice that tends to make tests harder to understand (and doesn't scale to testing non-trivially-constructed objects anway). You can just put:

```
{{pluginDartClass}} {{pluginClassLowerCamelCase}} = {{pluginDartClass}}();
```

at the start of each test case (in this case, all one of them).